nano/node/json_handler.cpp

```diff
@@ -4785,7 +4748,32 @@ void nano::json_handler::work_generate ()
 		{
 			ec = nano::error_rpc::difficulty_limit;
 		}
-		if (!ec)
+		// Retrieving optional block
+		std::shared_ptr<nano::block> block;
+		boost::optional<std::string> block_text (request.get_optional<std::string> ("block"));
+		if (!ec && block_text.is_initialized ())
```

Comment:
Could do `if (!ec && request.count ("block"))` to avoid retrieving the contents here.

Reply:
Thanks